src/python/pants/backend/python/lint/mypy/rules.py

```diff
@@ -77,17 +78,19 @@ async def mypy_lint(
     )
 
     prepared_sources_request = Get(
-        StrippedPythonSources,
-        StrippedPythonSourcesRequest(transitive_targets.closure, include_resources=False),
+        UnstrippedPythonSources,
+        UnstrippedPythonSourcesRequest(transitive_targets.closure, include_resources=False),
     )
     pex_request = Get(
         Pex,
         PexRequest(
             output_filename="mypy.pex",
             requirements=PexRequirements(mypy.get_requirement_specs()),
-            # TODO(#10131): figure out how to robustly handle interpreter constraints. Unlike other
-            #  linters, the version of Python used to run MyPy can be different than the version of
-            #  the code.
+            # NB: This only determines what MyPy is run with. The user can specify what version
+            # their code is with `--python-version`. See
+            # https://mypy.readthedocs.io/en/stable/config_file.html#platform-configuration. We do
+            # not auto-configure this for simplicity and to avoid Pants magically setting values for
+            # users.
```

Comment:
This is after discussion with Benjy yesterday.

There would be some complexities if Pants were to try to auto-configure `--python-version`:
1. Difficult to actually do. MyPy expects a 2-digit number like `3.6`, but our interpreter constraints express ranges like `>3.6`. It's not clear how to translate a more general constraint into something more specific.
2. We would need to then validate that the user did not try setting this and is now conflicting with us. Sounds leaky.